goloose.go

```diff
@@ -345,7 +348,10 @@ func customJson(in, out reflect.Value) (bool, error) {
 	}
 	inOk := in.Type().Implements(jsonMarshalerType)
 	outOk := out.Addr().Type().Implements(jsonUnmarshalerType)
-	if inOk || outOk {
+	// civil.Date implements encoding.TextUnmarshaler instead
+	isCivilDate := out.Type() == civilDateType
```

Comment:
since UnmarshalJson falls back to UnmarshalText I'm wondering if it's safe to check for `encoding.TextUnmarshaler` in a more generic way

Reply:
Yes, this package should be simulating encoding/json's behaviour! Pushed https://github.com/reillywatson/goloose/commit/0816d5ef574d6eb4b4d9d841095745b4ec2e0214 to handle this generically, so we don't need to depend on cloud.google.com/go, which is a pretty giant module (as the diffstat on go.sum in here indicates :D). Let me know if that works ok for you! It's quite similar to this change, only without the fastPath parts.

Reply:
Yes, works for me! Thanks :D